src/Test/Plutip/Contract.hs

```diff
@@ -330,14 +345,51 @@ instance
 
   testOptions = Tagged []
 
+{- | Run a contract using the first wallet as own wallet, and return ExecutionResult.
+ This could be used by itself, or combined with multiple other contracts.
+
+ @since 0.2
+-}
+withContract ::
+  forall (w :: Type) (s :: Row Type) (e :: Type) (a :: Type).
+  TestContractConstraints w e a =>
+  ([PaymentPubKeyHash] -> Contract w s e a) ->
+  TestRunner w e a
+withContract toContract = do
```

Comment:
Could it be expressed in terms of `withContractAs`? (looks like one of the cases of it)

Reply:
good catch